src/verification/tests.rs

```diff
@@ -47,16 +100,125 @@ fn test_jose_header() {
+                    "{\"alg\":\"RS256\",\"typ\":\"NOT_A_JWT\"}",
+                )
+                .expect("failed to deserialize"),
+            )
+            .expect("JWT type should be allowed but is not");
+
+        custom_verifier
+            .validate_jose_header(
+                &serde_json::from_str::<CoreJsonWebTokenHeader>(
+                    "{\"alg\":\"RS256\",\"typ\":\"application/NOT_A_JWT\"}",
+                )
+                .expect("failed to deserialize"),
+            )
+            .expect("JWT type should be allowed but is not");
+
+        assert_unsupported(
+            custom_verifier.validate_jose_header(
+                &serde_json::from_str::<CoreJsonWebTokenHeader>(
+                    "{\"alg\":\"RS256\",\"typ\":\"application/NOT_A_JWT;bla=test\"}",
+                )
+                .expect("failed to deserialize"),
+            ),
+            "unsupported JWT type",
+        );
+
+        custom_verifier
+            .validate_jose_header(
+                &serde_json::from_str::<CoreJsonWebTokenHeader>(
+                    "{\"alg\":\"RS256\",\"typ\":\"application/at+jwt\"}",
+                )
+                .expect("failed to deserialize"),
+            )
+            .expect("JWT type should be allowed but is not");
+
+        assert_unsupported(
+            custom_verifier.validate_jose_header(
+                &serde_json::from_str::<CoreJsonWebTokenHeader>(
+                    "{\"alg\":\"RS256\",\"typ\":\"NOT_A_JWT_REALLY\"}",
+                )
+                .expect("failed to deserialize"),
+            ),
+            "unsupported JWT type",
+        );
+    }
+
+    // Allow all JWT types.
+    {
+        let custom_verifier = CoreJwtClaimsVerifier::new(
+            client_id.clone(),
+            issuer.clone(),
+            CoreJsonWebKeySet::new(vec![]),
+        )
+        .allow_all_jose_types();
+
+        custom_verifier
+            .validate_jose_header(
+                &serde_json::from_str::<CoreJsonWebTokenHeader>(
+                    "{\"alg\":\"RS256\",\"typ\":\"NOT_A_JWT\"}",
+                )
+                .expect("failed to deserialize"),
+            )
+            .expect("JWT type should be allowed but is not");
+
+        custom_verifier
+            .validate_jose_header(
+                &serde_json::from_str::<CoreJsonWebTokenHeader>(
+                    "{\"alg\":\"RS256\",\"typ\":\"application/at+jwt\"}",
+                )
+                .expect("failed to deserialize"),
+            )
+            .expect("JWT type should be allowed but is not");
+
+        custom_verifier
+            .validate_jose_header(
+                &serde_json::from_str::<CoreJsonWebTokenHeader>(
+                    "{\"alg\":\"RS256\",\"typ\":\"NOT_A_JWT_REALLY\"}",
+                )
+                .expect("failed to deserialize"),
+            )
+            .expect("JWT type should be allowed but is not");
+    }
```

Comment:
let's add one more case with a parameterized MIME type (i.e., with a `;`) since those should be accepted here

Reply:
have you had a chance to add this test case?

Reply:
Ah sorry, I did not read exactly and thought you meant another case. Now a test case should be added.